packages/theme/src/document-head-contents/utils/generate-css-color-variables.ts

```diff
@@ -0,0 +1,36 @@
+const toRgb = (color: string) => {
+  switch (true) {
+    case /^#[\da-fA-F]{3}$/.test(color):
+      return [
+        color.slice(1, 2) + color.slice(1, 2),
+        color.slice(2, 3) + color.slice(2, 3),
+        color.slice(3, 4) + color.slice(3, 4),
+      ].map((val) => parseInt(val, 16))
+    case /^#[\da-fA-F]{6}$/.test(color):
+      return [color.slice(1, 3), color.slice(3, 5), color.slice(5, 7)].map(
+        (val) => parseInt(val, 16)
+      )
+    case /^rgb\(\s*\d+\s*,\s*\d+\s*,\s*\d+\s*\)$/.test(color):
+      return color
+        .replace(/^.*\((.*)\).*$/, '$1')
+        .split(',')
+        .map((n) => parseInt(n))
+        .slice(0, 3)
+    default:
+      return null
+  }
+}
+
+export const generateCssColorVariables = (
```

Comment:
Maybe add docs/examples/tests what this function do? It feels that this function is important

Reply:
how the tests should look like? no idea tbh